mypy/test/testcmdline.py

```diff
@@ -50,7 +51,8 @@ def test_python_cmdline(testcase: DataDrivenTestCase, step: int) -> None:
     args = parse_args(testcase.input[0])
     custom_cwd = parse_cwd(testcase.input[1]) if len(testcase.input) > 1 else None
     args.append('--show-traceback')
-    args.append('--legacy')
+    if 'based' not in testcase.file.rsplit(os.sep)[-1]:
+        args.append('--legacy')
```

Comment:
i think it's confusing magic behavior to use the file names to change the behavior of the tests

Reply:
I agree, but it's consistent with the current setup.